packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -68,11 +68,22 @@ export function mediaUpload( {
 
 	// Allowed type specified by consumer
 	const isAllowedType = ( fileType ) => {
-		return ( allowedType === '*' ) || startsWith( fileType, `${ allowedType }/` );
+		return ( ! allowedTypes ) ||
+			some(
+				allowedTypes,
+				( allowedType ) => {
```

Comment:
Alternatively this could be extracted into a separate function for checking a single allowedType.